### PR TITLE
[Sensor] Fisheye and Equirectangular semantic sensors

### PR DESCRIPTION
**Pull request by bigbike**
## Motivation and Context
This diff does the following stuffs:
- developed 2 new sensors: fisheye semantic sensor, and equirectangular semantic sensor;
- developed new visualization functionality to visualize the semantic info from the aforementioned sensors;
- added demos in the utility viewer (We chose parameters for "GoPro" in our demo);
- small fixes;

The results are as follows:
fe-color
fe-semantic
eRec-color
eRec-semantic

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bigbike** [comment]
@mosra: No kidding. I have no idea why the stupid github cannot let me add you as a reviewer in this diff. Literally cannot find you in the list. Would you add yourself?

File: src/esp/sensor/CubeMapSensorBase.cpp [hunk 1]
```diff
@@ -129,10 +135,11 @@ bool CubeMapSensorBase::renderToCubemapTexture(sim::Simulator& sim) {
     if (twoSceneGraphs) {
       flags |= gfx::RenderCamera::Flag::ObjectsOnly;
       // Incremental rendering:
-      // BE AWARE that here "ClearColor" and "ClearDepth" is NOT set!!
-      // Rendering happens on top of whatever existing there.
+      // BE AWARE that here "ClearColor", "ClearDepth" and "CLearObjectId" are
```
Comment (Skylion007) [inline src/esp/sensor/CubeMapSensorBase.cpp:138]: Typo

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
@bigbike This ended up being a bit more involved so I added the tests myself.

**Comment by erikwijmans** [comment]
@bigbike @Skylion007 looks like this is going to revert #1222 ?

**Comment by bigbike** [comment]
> @erikwijmans Fixed.
> 
> @bigbike Your changes reminded me that the SensorSpec has a channels property that weren't setting with Python or Habitat-Lab so I tried to update the example to set it. Interestingly, whenever I see it to 3 channels it crashes.

This is because the default value for channel should be 4 not 3 because there is an `alpha` channel. See here:
https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/sensor/VisualSensor.h#L35-L38

**Comment by Skylion007** [comment]
LGTM: Let's merge.

File: examples/settings.py [hunk 2]
```diff
@@ -17,11 +17,15 @@
     "color_sensor": True,  # RGB sensor (default: ON)
     "semantic_sensor": False,  # semantic sensor (default: OFF)
     "depth_sensor": False,  # depth sensor (default: OFF)
-    "ortho_sensor": False,  # Orthographic RGB sensor (default: OFF)
+    "ortho_rgb_sensor": False,  # Orthographic RGB sensor (default: OFF)
```
Comment (Skylion007) [inline examples/settings.py:20]: @bigbike This should probably be renamed to rgba. I wasn't aware they returned RGBA info until now.

**Comment by Skylion007** [comment]
@bigbike It all passed except for that flakey codecov upload.

**Comment by bigbike** [comment]
@Skylion007 : I re-ran the CI. It passed all the tests. Now it is ready to be committed. THANK YOU very much for contributing to this big PR as well! I sincerely appreciate it.

Merge it in.

**Pull request merged**